source/internal/_reduce.js

```diff
@@ -30,13 +32,30 @@ function _iterableReduce(xf, acc, iter) {
   return xf['@@transducer/result'](acc);
 }
 
+function _objectReduce(xf, acc, object) {
+  for (var key in object) {
+    if (_has(key, object)) {
+      acc = xf['@@transducer/step'](acc, object[key], key);
+      if (acc && acc['@@transducer/reduced']) {
+        acc = acc['@@transducer/value'];
+        break;
+      }
+    }
+  }
+  return xf['@@transducer/result'](acc);
+}
+
 function _methodReduce(xf, acc, obj, methodName) {
   return xf['@@transducer/result'](obj[methodName](bind(xf['@@transducer/step'], xf), acc));
 }
 
 var symIterator = (typeof Symbol !== 'undefined') ? Symbol.iterator : '@@iterator';
+var typeErrorMessage = 'reduce: list must be array or iterable';
```

Comment:
```suggestion
var typeErrorMessage = 'reduce: input must be object or iterable';
```

Reply:
Hi! I don't like the current error message, but I think the proposal isn't accurate enough when `_reduce(R.take(1), {}, {a: 1, b: 2})` throws an error and `_reduce(R.map(x => 2 * x), {}, {a: 1, b: 2})` doesn't.

Reply:
```js
  if (_isObject(list)) {
    if (fn['@@transducer/commutative']) {
      return _objectReduce(fn, acc, list);
    } else {
      throw new TypeError('reduce: passed transducer requires ordered input');
      // or
      throw new TypeError('reduce: unordered input passed for ordered transducer');
    }
  }
```